### DISCUSSION TITLE
Copy url button

### DISCUSSION THREAD

**Discussion by ugintl** [JulianPrieber/littlelink-custom#157] [General]
See the screenshot. I want to show  copy button to copy the whole profile link. How to do it? I have tried different things, but not able to do it.

![copy-button](https://user-images.githubusercontent.com/12570802/189480820-e345aaf7-bb80-4df0-8361-8d92e290aed2.jpeg)

**Comment by JulianPrieber** [comment]
Making a button to copy the URL is fairly easy.
For this, the code on the littlelink page for the share button can be used.

Optimally, you want some form of feedback that the URL has been copied like you see on: https://littlelink-custom.com/docker

**Reply by JulianPrieber** [reply]
Hmm, I'm not sure anymore.

I'm not sure if you could copy the profile URL with the share button code (https://github.com/JulianPrieber/littlelink-custom/blob/main/littlelink/js/share.button.js).

I'm not really good working with JS. Optimally, you'd have to get this working with inline JS, so you can use blade or PHP to get the profile URL.

`<?php echo url('') . "/@" . $page->littlelink_name; ?>`

**Reply by JulianPrieber** [reply]
@ugintl I think I got it.

It's a bit janky, but this should work:
```
<?php $fullURL = url('') . "/@" . $page->littlelink_name; 
echo "<button onclick=\"copyToClipboard(" . "'" . $fullURL . "'" . ")\">Copy to Clipboard</button>"
?>

<script>
  function copyToClipboard(text) {
    var dummy = document.createElement("textarea");
    document.body.appendChild(dummy);
    dummy.value = text;
    dummy.select();
    document.execCommand("copy");
    document.body.removeChild(dummy);
    alert("Done!");
  }
</script>
```

**Reply by ugintl** [reply]
It worked thanks. Your solution is quite different than what I was trying. However, the page is refreshing after clicking "copy" button. I will try to improve it and share code here if i am successful. Thanks again.

**Reply by JulianPrieber** [reply]
The page should definitely not be refreshing when pressing the button.
There should only be the alert.
